pkg/schemalog/log_entry.go

```diff
@@ -64,8 +64,6 @@ func (m *LogEntry) UnmarshalJSON(b []byte) error {
 			if err := json.Unmarshal([]byte(schemaStr), &m.Schema); err != nil {
 				return err
 			}
-		default:
```

Comment:
Why do we need the custom `UnmarshalJSON` method? It doesn't seem to be doing anything special apart from the `schema` case where it checks the length. If this is the only special case then it could be moved into some code on the `Schema` type.

Reply:
To be honest, I'm not sure why we needed this. I will have a look and see if we can get rid of it (will open a separate PR for that if needed) 👍

Reply:
I was looking into this, we need the custom unmarshal because we use it to unmarshal from the WAL event, from a map[string]any representation instead of a log entry json one. 
https://github.com/xataio/pgstream/blob/main/pkg/wal/processor/wal_processor.go#L51-L53

Reply:
If that's the only place then it would probably be better to do the conversion in `WalDataToLogEntry` by looking at the column name / values and populating the LogEntry there rather than having that logic buried in the JSON decoder.

Not urgent, just feels like a bit of tech debt.